TLM/CSUtil.Commons/VectorUtil.cs

```diff
@@ -1,22 +1,27 @@
-﻿using System;
-using System.Collections.Generic;
-using System.Linq;
-using System.Text;
-using UnityEngine;
+﻿namespace CSUtil.Commons {
+    using UnityEngine;
 
-namespace CSUtil.Commons {
-	public static class VectorUtil {
-		public static void ClampRectToScreen(ref Rect rect, Vector2 resolution) {
-			Log._Debug($"ClampPosToScreen([{rect.x}, {rect.y}, {rect.xMax}, {rect.yMax}], [{resolution.x}, {resolution.y}]) called");
-			if (rect.x < 0)
-				rect.x = 0;
-			if (rect.y < 0)
-				rect.y = 0;
-			if (rect.xMax >= resolution.x)
-				rect.x = resolution.x - rect.width;
-			if (rect.yMax >= resolution.y)
-				rect.y = resolution.y - rect.height;
-			Log._Debug($"ClampPosToScreen() -> [{rect.x}, {rect.y}, {rect.xMax}, {rect.yMax}]");
-		}
-	}
-}
+    public static class VectorUtil {
+        public static void ClampRectToScreen(ref Rect rect, Vector2 resolution) {
+            // Log._Debug($"ClampPosToScreen([{rect.x}, {rect.y}, {rect.xMax}, {rect.yMax}], " +
+            //           $"[{resolution.x}, {resolution.y}]) called");
+            if (rect.x < 0) {
+                rect.x = 0;
+            }
+
+            if (rect.y < 0) {
+                rect.y = 0;
+            }
+
+            if (rect.xMax >= resolution.x) {
+                rect.x = resolution.x - rect.width;
+            }
```

Comment:
Would this yield a small performance increase:

```csharp
if (rect.xMax >= resolution.x) {
    rect.x = resolution.x - rect.width;
} else if (rect.x < 0) {
    rect.x = 0;
}
```

That way, if the UI is found to be outside right bounds (most likely bounds issue I think), there's no need to check the left bounds.

Same could be done for `rect.y`.

Reply:
If this code is only used for UI, then it will run a few times per window move, that is not critical. I would consider something using Min/Max, like `rect.x = Mathf.Max(0f, Mathf.Min(rect.x, resolution.x - rect.width)` it is shorter but performs subtraction regardless of actual need in it. Your variant can be a tiny bit faster if one of the screen edges is touched.

Reply:
Replaced with 
```
rect.x = Mathf.Clamp(rect.x, 0f, resolution.x - rect.width);
rect.y = Mathf.Clamp(rect.y, 0f, resolution.y - rect.height);
```